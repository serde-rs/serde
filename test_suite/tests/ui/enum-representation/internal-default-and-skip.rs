use serde_derive::Deserialize;

#[derive(Deserialize)]
#[serde(tag = "tag")]
enum E1 {
    V1 { f: u8 },
    #[serde(default, skip)]
    V2 { f: u8 },
}

#[derive(Deserialize)]
#[serde(tag = "tag")]
enum E2 {
    V1 { f: u8 },
    #[serde(default, skip_deserializing)]
    V2 { f: u8 },
}

fn main() {}
