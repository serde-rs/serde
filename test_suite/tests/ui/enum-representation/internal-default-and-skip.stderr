error: variant `V2` cannot have both #[serde(default)] and #[serde(skip)] or #[serde(skip_deserializing)]
 --> tests/ui/enum-representation/internal-default-and-skip.rs:7:5
  |
7 | /     #[serde(default, skip)]
8 | |     V2 { f: u8 },
  | |________________^

error: variant `V2` cannot have both #[serde(default)] and #[serde(skip)] or #[serde(skip_deserializing)]
  --> tests/ui/enum-representation/internal-default-and-skip.rs:15:5
   |
15 | /     #[serde(default, skip_deserializing)]
16 | |     V2 { f: u8 },
   | |________________^
